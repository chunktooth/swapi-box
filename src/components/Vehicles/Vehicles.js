import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';

const Vehicles = ({data, setFavorites}) => {
	const displayVehicles = data.map((vehicle, index) => {
		return <Card
			key={index} 
			data={vehicle} 
			setFavorites={setFavorites}
		/>
	});

	return (
		<article>
			{displayVehicles}
		</article>
	)
}

export default Vehicles;