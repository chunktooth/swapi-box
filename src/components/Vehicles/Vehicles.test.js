import React from 'react';
import Vehicles from './Vehicles';
import {Shallow} from 'enzyme';

describe('Vehicles', () => {
	let vehicles;

	beforeEach(() => {
		vehicles = shallow(<Vehicles />);
	});

	it('should match the snapshot', () => {
		expect(vehicles).toMatchSnapshot();
	});
	
})