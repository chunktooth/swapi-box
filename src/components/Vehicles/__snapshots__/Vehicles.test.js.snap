// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    data={
        Array [
            Object {
              "class": "Demolition",
              "model": "Tank",
              "name": "El Toro Loco",
              "passengers": "1",
            },
          ]
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          data={
                    Object {
                              "class": "Demolition",
                              "model": "Tank",
                              "name": "El Toro Loco",
                              "passengers": "1",
                            }
          }
          setFavorites={[Function]}
/>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "class": "Demolition",
            "model": "Tank",
            "name": "El Toro Loco",
            "passengers": "1",
          },
          "setFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            data={
                        Object {
                                    "class": "Demolition",
                                    "model": "Tank",
                                    "name": "El Toro Loco",
                                    "passengers": "1",
                                  }
            }
            setFavorites={[Function]}
/>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "class": "Demolition",
              "model": "Tank",
              "name": "El Toro Loco",
              "passengers": "1",
            },
            "setFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
